src/ui/CategoryFacet/CategoryFacet.ts

```diff
@@ -763,12 +797,22 @@ export class CategoryFacet extends Component implements IAutoLayoutAdjustableIns
   private initQueryStateEvents() {
     this.queryStateModel.registerNewAttribute(this.queryStateAttribute, this.options.basePath);
     this.bind.onQueryState<IAttributesChangedEventArg>(MODEL_EVENTS.CHANGE, undefined, data => this.handleQueryStateChanged(data));
+    this.dependsOnManager.listenToParentIfDependentFacet();
+  }
+
+  private initDependsOnManager() {
+    this.dependsOnManager = new DependsOnManagerUtils(this, () => this.changeActivePath(this.options.basePath));
```

Comment:
Hmm, for the second arg to the constructor, since we are passing `this`, I think would it be cleaner to create an interface:
```
export interface IDependentFacet {
    reset: () => void
}
```

and then the CategoryFacet, Facet implement this interface. Inside the `DependsOnManagerUtils`, we just need to call the same method specified in the interface. You may need to add more properties to the interface depending on what is used inside of the `DependsOnManagerUtils` class.

Moreover, for typing the `this` param, rather than doing:

```
constructor(private facet: CategoryFacet | Facet)
```
we could say we expect an object of type `IDependentFacet `. This should make it easier to unit test, and if we add the option to other facets in the future, no need to keep adding `| <facet> | <another facet>`.

Reply:
Good Idea, I'll work something out.

Reply:
Where should I declare the interface so that it can be used by facet, categoryfacet and the dependsOnManager?

Reply:
I think it makes the most sense to include in the `DependsOnManager` file